### ISSUE TITLE
Zootaxa - style change request

### ISSUE THREAD

**Issue by proceps** [citation-style-language/styles#3769]
is it possible to add a number after the volume and stated_year before the the volume if those values are present in BibTex for Zootaxa style. In zoological practice people pay close attention to the actual publication date of the article. Actual year of publication is very important in zoological nomenclature. So people cite both actual year of publication (first) and stated_year (second if it is different from the first one).

Kiss, O. (2013) Three new species of Rhyacophila (Trichoptera: Rhyacophilidae) from Taiwan and Nepal. Zootaxa, **(2012),** 3640 **(2)**, 213–223.

**Comment by adam3smith** [comment]
stated_year doesn't exist in CSL, so that's definitely not possible, but I also don't see why we would add that or the issue number to the Zootaxa style. They're not in the journal style guide (which seems pretty thorough) and they are not in any references printed by the journal that I could find -- so while I'm not a Zoologist and can't judge what's common in your field, it's simply not the case that that's how people cite in the journal Zootaxa.

**Comment by proceps** [comment]
this is Zootaxa reference guide:
http://mapress.com/zootaxa/support/reference.pdf
The number is there.

**Comment by adam3smith** [comment]
This is the reference guide, too, and it doesn't have issue numbers http://www.mapress.com/j/zt/pages/view/forauthors nor do published articles (or at least the ones I could access). Do you have any insight information from the journal they'd want this or what's the source of this request.

**Comment by proceps** [comment]
this paper, published just few days ago have both: volume numbers and both actual and stated years of publication:
http://www.mapress.com.proxy2.library.illinois.edu/j/zt/article/view/zootaxa.4507.1.1

**Comment by proceps** [comment]
![image](https://user-images.githubusercontent.com/5597748/47958868-a920df80-dfa2-11e8-9149-9bdc2e310820.png)

**Comment by adam3smith** [comment]
And this one, also just published, doesn't http://www.mapress.com/j/zt/article/view/zootaxa.4508.1.6
I really don't have a dog in this fight, but I don't see a strong case for changing the style. Why are you so adamant about this?

**Comment by proceps** [comment]
It is not about fight, it is about providing all information, which is available in the database, and not twicking output manually, because some of the information was omitted. And probably improving the style.

**Comment by proceps** [comment]
and in two different examples you can see that there are people who do not care, and those who care.

**Comment by rmzelle** [comment]
@proceps, can you clarify for me what "stated_year" means exactly? Is it different from e.g. the online-first date of articles that first appear online and are later assigned an issue? Is it e.g. for articles in journal January issues that already become available in December of the previous year?

And are you aware of any other journals in zoology that have a clearer requirement for "stated_year"?

**Comment by proceps** [comment]
you have ZooKeys, so it have the same practice. the stated_year is appropriate in the journals which deal with the description of new taxa. Lets take your example. Most usual is the opposite situation. The December issue of 1999 became available only in January 2000. So we have bibtex.year = 2000 and bibtex.stated_year = 1999. If a new species is described in this journal, it will be cited as "Aus bus Smith, 2000", because of the importance when this species name actually became available for other researchers. So when we cite the species, we will cite it as Smith 2000, and of cause, in the list of bibliographic references we will have Smith 2000 to match. The problem is that the journal itself does not show 2000 on the title page, it shows 1999 (only inside you can find actual date of publication in a small font). For older journals, not rare you can find, that each article was published on different date. So, to make a proper citation, when both dates are available, and when both do not match, author include the "state_year", in this journal in square brackets after bibliography (the way I showed in the example above). Myself more used to the style when that stated year is included before the volume, but it does not matter. In general it is relatively rare cases, may be 5% of the citation which have two dates. I can take a look what other journals you have on the list. But the most important are Zootaxa and ZooKeys, that is where most of the systematic related papers are published now days. For ecological, molecular and other papers in zoology, people usually use only one date.

**Comment by rmzelle** [comment]
Thanks. I looked through the author's instructions for ZooKeys, but also there I couldn't find any mention of citing both the stated_year and regular publication date. I don't doubt that people do what you describe, but it would be easier to make the argument for adding a "stated_year" CSL date variable if we could point to some official citation guidelines. Please let us know if you come across any.

**Comment by proceps** [comment]
The most important source is the Code of Zoological Nomenclature (ICZN). Which provide guidance how to describe and handle zoological name. Including citations of zoological names. Specifically article 22 of the Code:
http://www.nhm.ac.uk/hosted-sites/iczn/code/index.jsp?nfv=true&article=22
It is very rare, that the journals put this in their reference guidance, they provide only basic rules.
I found one journal actually put it in the guideline:
Journal of the Kansas Entomological Society:
http://kes.allenpress.com/kes/portals/_default/instructions_for_authors.pdf

**Comment by rmzelle** [comment]
@proceps, thanks, that's very useful.

http://www.nhm.ac.uk/hosted-sites/iczn/code/index.jsp?nfv=true&article=22

> 22A.2. Method of citation. In citing the date of publication of a name, an author
> ... 
> 22A.2.2. if the actual date of publication is different from the date specified in the work (imprint date), should cite the actual date of publication; except that
> 22A.2.3. if wishing to cite both the actual and the imprint dates, should first cite the actual date (cited as above), followed by the imprint date for information and enclosed in parentheses or other brackets and quotation marks; ...
>
> Examples. Ctenotus alacer Storr, 1970 ("1969"), or Ctenotus alacer Storr, 1970 ["1969"], or Ctenotus alacer Storr, 1970 (imprint 1969), or Ctenotus alacer Storr, 1970 (not 1969), was established in a work which, although published in 1970, carried an imprint date of 1969; Anomalopus truncatus (Peters, 1876 ["1877"]) was established in a different genus from Anomalopus in a work which, although published in 1876, carried an imprint date of 1877.



http://kes.allenpress.com/kes/portals/_default/instructions_for_authors.pdf

> For articles that were printed with one year but did not appear or were not actually issued
until a subsequent year, cite the correct year in brackets following the year as printed on the
original publication. Refer to the printed year for all citations in the text. If more precise
information on dating of a particular article is needed (e.g., for purposes of taxonomic priority),
this can be provided in brackets following the citation.
>
>EXAMPLES:
>Darling, D. C. 1996 [1997]. A new species of Spalangiopelta (Hymenoptera: Pteromalidae; Ceinae) from Dominican amber: Phylogenetic and biogeographic implications. Journal of the Kansas Entomological Society, supplement 69(4): 248-259.
> Moure, J. S., and J.M.F. Camargo. 1978. A fossil stingless bee from copal (Hymenoptera: Apidae). Journal of the Kansas Entomological Society 51(4): 550-566. [Publ. date: 17 November 1978]

As we mentioned above, CSL currently doesn't have the capability to store the "imprint" date, but we'll see if we can make this possible in a future version.

**Comment by adam3smith** [comment]
I think we should move this to a schema ticket since we don't have stale date. Reopening until we have this in a new ticket.

**Comment by rmzelle** [comment]
@adam3smith, @proceps, I just created a follow-up issue at https://github.com/citation-style-language/schema/issues/163. I suggest closing this issue.

Thinking about how to implement this, I did just realize that we might need a test to determine whether the imprint date and actual date of publication fall in different years, since this generally seems to be a requirement for showing both dates.

Such a test can either be hardcoded and implicit (i.e. calling the "imprint" date in the style will only return a value if the date is in a different year than the "issued" date), but we currently don't use such logic anywhere and it's less flexible. Alternatively we probably would have to extend `<choose/>` with a date-comparing conditional (e.g. `<if is-different-years="issued imprint">`).
